ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/date/DateTaskFragment.kt

```diff
@@ -28,16 +30,21 @@ import com.google.android.ground.ui.datacollection.tasks.AbstractTaskFragment
 import dagger.hilt.android.AndroidEntryPoint
 import java.util.*
 import kotlinx.coroutines.flow.MutableStateFlow
-import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.map
 import org.jetbrains.annotations.TestOnly
 
 @AndroidEntryPoint
 class DateTaskFragment : AbstractTaskFragment<DateTaskViewModel>() {
 
   private var datePickerDialog: DatePickerDialog? = null
 
-  private var _dateText: MutableStateFlow<String> = MutableStateFlow("")
-  val dateText = _dateText.asStateFlow()
+  private val _timestamp: MutableStateFlow<Date?> = MutableStateFlow(null)
```

Comment:
We can reuse the taskData from view model instead.

Reply:
viewmodel might fail to access, let me check

Reply:
You can do something like this:

```
  lateinit var dateText: LiveData<String>

  override fun onTaskViewAttached() {
    super.onTaskViewAttached()
    dateText =
      viewModel.taskTaskData
        .filterIsInstance<DateTimeTaskData?>()
        .map { taskData ->
          if (taskData != null) {
            val calendar = Calendar.getInstance()
            calendar.timeInMillis = taskData.timeInMillis
            getDateFormatter().format(calendar.time) ?: ""
          } else {
            ""
          }
        }
        .asLiveData()
  }
  ```

Also, please test all that date/time task data also gets saved without clicking any buttons.